docs/superpowers/specs/2026-07-05-vite-component-rules-design.md

```
@@ -0,0 +1,282 @@
+`import { collectComponentFacts } from './providers/source/components.js';`,
+call `const components = await collectComponentFacts(cwd);` alongside the
+existing `collectRenderedProject` call, and add `components` to the
+`runRules` context: `{ heads, headings, images, project, components, config
+}`. This call is unconditional — independent of whether any prerendered pages
+exist, since it doesn't touch `prerenderPagesDir` — so it runs even in the
+"0 routes" case (that early-exit already happens in `plugin.ts`, after
+`analyze()` returns, for reporting/gating purposes only, not before).
+
+Extend the `coverageNote` with a second line:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)